api/app/models/spree/api/webhooks/stock_item_decorator.rb

```diff
@@ -0,0 +1,48 @@
+module Spree
+  module Api
+    module Webhooks
+      module StockItemDecorator
+        def self.prepended(base)
+          base.around_save :queue_webhooks_requests_for_variant_back_in_stock!
+          base.around_save :queue_webhooks_requests_for_variant_backorderable!
+        end
+
+        private
+
+        def queue_webhooks_requests_for_variant_back_in_stock!
+          was_out_of_stock = !variant_in_stock?
+          yield
+          if was_out_of_stock && variant_in_stock?
+            variant.queue_webhooks_requests!('variant.back_in_stock')
+          end
+        end
+
+        def queue_webhooks_requests_for_variant_backorderable!
+          was_out_of_stock = !variant_in_stock?
+          was_not_backorderable = !variant_backorderable?
+          yield
+          touch # changes must be reflected before instantiating the serializer
+          return unless emits_webhook_event?(was_not_backorderable, was_out_of_stock)
+
+          variant.queue_webhooks_requests!('variant.backorderable')
+        end
+
+        def emits_webhook_event?(was_not_backorderable, was_out_of_stock)
+          was_out_of_stock && was_not_backorderable && variant_in_stock? && variant_backorderable?
+        end
+
+        # rewriting `variant.in_stock?` as is currently being cached
+        def variant_in_stock?
+          Spree::Variant.in_stock.exists?(id: variant.id)
```

Comment:
```suggestion
          variant.stock_items.where('count_on_hand > 0').or(variant.stock_items.where(backourderable: true)
```

this will eliminate a lot of SQL queries in the process

Reply:
Shouldn't we take into account the `should_track_inventory = false` condition? I took it as an example from [here](https://github.com/spree/spree/blob/d5eda640caf7b84fd75cd83846048b5efa07ec9e/core/app/models/spree/variant.rb#L68).

Reply:
@sebastian-palma yes we should, good catch!

Reply:
Having both checks in a single place kind of complicates the process, because updating the backorderable and count_on_hand attributes must be done in two separate operations (if using `set_count_on_hand`), and it never returns true when asking if the variant was previously out of stock. What's the "production" workflow for making a variant back in stock @damianlegawiec so we can test it right the same way?

Reply:
Made some changes based on the [Admin Dashboard workflow](https://github.com/spree/spree_backend/blob/main/app/controllers/spree/admin/stock_items_controller.rb#L13:L26) for updating stock items `count_on_hand`'s.